src/query/sql/src/planner/metadata.rs

```diff
@@ -402,12 +400,13 @@ impl Metadata {
                     field.data_type().clone(),
                     table_index,
                     path_indices,
-                    None,
+                    Some(field.column_id),
                     None,
                     None,
                 );
 
                 let mut i = fields_type.len();
+                let mut inner_column_id = field.column_id;
                 for (inner_field_name, inner_field_type) in
                     fields_name.iter().zip(fields_type.iter()).rev()
```

Comment:
Remove `rev()` here makes the `fields_name` and `fields_type` loop from left to right, and changes `i` to incrementing from 0, because we don't need to compute the `leaf_index` anymore, and the inner field can iterate in the normal order.

Reply:
Will this cause a breaking change ?

Reply:
will not cause a breaking change, `leaf_index` is not used anymore.